aten/src/ATen/native/cuda/Nonzero.cu

```diff
@@ -137,4 +165,132 @@ Tensor nonzero_cuda(const Tensor& self){
   Tensor out = at::detail::empty_cuda({0}, self.options().dtype(kLong));
   return at::native::nonzero_out_cuda(self, out);
 }
+
+template <typename scalar_t>
+void nonzero_static_cuda_out_impl(
+    const Tensor& self,
+    int size,
+    int64_t fill_value,
+    Tensor& out) {
+  Tensor self_ = self.contiguous();
+  const cudaStream_t stream = at::cuda::getCurrentCUDAStream();
+  // compute number of nonzero elements
+  size_t temp_storage_bytes = 0;
+  auto& allocator = *c10::cuda::CUDACachingAllocator::get();
+  auto num_nonzeros = allocator.allocate(sizeof(int));
+  cub::TransformInputIterator<bool, NonZeroOp<scalar_t>, const scalar_t*> itr(
+      self_.const_data_ptr<scalar_t>(), NonZeroOp<scalar_t>());
+  // expected output size is num_nonzeros x ndim
+  // we are producing output with size {num_nonzeros, ndim} and strides {1,
+  // num_nonzeros} (that is, transposed ndim x num_nonzeros output) we are able
+  // to directly use passed output with this size and strides, and we can also
+  // (per contract) resize passed output with incorrect sizes anyway we want.
+  // However, out with correct sizes and incorrect strides will have to be
+  // copied to from the intermediate we've produced.
+  bool need_to_copy = out.dim() == 2 && out.sizes()[0] == size &&
+      out.sizes()[1] == self.dim() && !out.t().is_contiguous();
```

Comment:
this will always fail, unless you either allocate `out` with 0 size (like for the non-static case) or allocate it as a transposed tensor, and thus you'll incur extra copy